connectors/src/connectors/zendesk/temporal/activities.ts

```diff
@@ -434,10 +435,20 @@ export async function syncZendeskTicketBatchActivity({
 
   const users = await zendeskApiClient.users.list();
 
-  const res = await concurrentExecutor(
+  const comments = await concurrentExecutor(
     tickets,
-    async (ticket) => {
-      const comments = await zendeskApiClient.tickets.getComments(ticket.id);
+    async (ticket) => zendeskApiClient.tickets.getComments(ticket.id),
+    { concurrency: 3, onBatchComplete: heartbeat }
+  );
+
+  const res = await concurrentExecutor(
+    _.zip(tickets, comments),
+    async ([ticket, comments]) => {
+      if (!ticket || !comments) {
```

Comment:
Both are technically `| undefined` as there's no typescript guarantee that both arrays are equal size

Reply:
OOC: do you prefer this over a map over the tickets to turn them into objects?

Reply:
Shorter like this but I don't have a strong opinion really

Reply:
fair 👍